eclipse-collections/src/main/java/org/eclipse/collections/impl/set/sorted/mutable/TreeSortedSet.java

```diff
@@ -253,13 +253,7 @@ public boolean equals(Object object)
             return true;
         }
 
-        if (!(object instanceof Set))
-        {
-            return false;
-        }
-
-        Set<?> other = (Set<?>) object;
-        return other.equals(this.treeSet);
+        return this.treeSet.equals(object);
```

Comment:
I'm not opposed to this change, but it was a weird sort of optimization. What is the speed of treeSet.equals(unifiedSet)? It could be linear or O(n log n). You iterate over one and call contains on the other.

Reply:
`equals` in object-oriented APIs has always felt wrong to me because of this kind of things, ideally `a.equals(b)` should be the same as `b.equals(a)`, but in reality, there could be some situation like here where one way is maybe better than the other (until it's not because you don't know the concrete implementation of `object` :)

thx for the review in any case!